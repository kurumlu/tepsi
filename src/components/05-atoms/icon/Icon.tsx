import * as React from 'react';
import { HTMLProps } from 'react';
import cn from 'classnames';
import { Sources, IconName } from '../../00-helpers/icons/icons';

export enum IconSize {
  EXTRA_MINOR = 'extra-minor',
  MINOR = 'minor',
  MEDIUM = 'medium',
  MAJOR = 'major',
  EXTRA_MAJOR = 'extra-major',
}

export enum IconColor {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  TERTIARY = 'tertiary',
  ACTION = 'action',
  ACTION_NAV = 'action-nav',
  NEGATIVE = 'negative',
  POSITIVE = 'positive',
  SUBTLE = 'subtle',
  NOTICE = 'notice',
}

export enum IconAnimation {
  BOP = 'bop',
  HEART_PULSE = 'heart-pulse',
  SPIN = 'spin',
}

export type IconProps = {
  name: IconName;
  id?: string;
  className?: string;
  title?: string;
  color?: IconColor;
  size?: IconSize;
  animation?: IconAnimation;
  dynamicContent?: string;
  ariaLabel?: string;
  ariaHidden?: boolean;
  inline?: boolean;
} & Pick<
  HTMLProps<HTMLSpanElement>,
  'onClick' | 'onMouseEnter' | 'onMouseLeave'
>;

export function Icon({
  name,
  id,
  className,
  title,
  color,
  size = IconSize.MEDIUM,
  animation,
  dynamicContent,
  ariaLabel,
  ariaHidden,
  inline,
  onClick,
  onMouseEnter,
  onMouseLeave,
}: IconProps) {
  const isInteractive =
    !!dynamicContent || !!onClick || !!onMouseEnter || !!onMouseLeave;

  const classes = cn(
    'sd-a-icon',
    { [`sd-a-icon--${name}`]: name },
    { [`sd-a-icon--${color}`]: color },
    { [`sd-a-icon--${size}`]: size },
    { [`sd-a-icon--animation-${animation}`]: animation },
    {
      [`sd-a-icon--interactive`]: isInteractive,
    },
    { [`sd-a-icon--inline`]: !!inline },
    className
  );

  const SourceComponent = Sources[name];

  return (
    <span
      role="img"
      id={id}
      className={classes}
      aria-label={ariaLabel}
      aria-hidden={ariaHidden}
      title={title}
      data-dynamiccontent={dynamicContent}
      onClick={onClick}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}>
      <SourceComponent
        focusable="false"
        aria-hidden="true"
        className="sd-a-icon__svg"
      />
    </span>
  );
}
export { IconName };
