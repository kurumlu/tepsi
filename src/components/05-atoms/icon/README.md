# Icon

#### How to add a new icon

Follow these steps:

1. Make sure the designer has added the icon in the [Figma library](https://www.figma.com/file/d3S8XD5WMrukNZQwFEWY9T/Library?node-id=2069%3A6503)
2. Go to the correct icon in the **Atoms/Icon/Large** icon set (all added icons must be 24x24)
3. Select the figma component, make sure to select the component, which is 24px by 24px and includes the bounding box of the icon, and not just the icon
4. Right click on the component and select "Copy as SVG"
5. Create a new file with the component name inside the raw icons directory: `src/components/00-helpers/icons/raw` 
for example: `src/components/00-helpers/icons/raw/star.svg` and paste the contents from figma inside this file
6. Run the `yarn generate-icon-exports` command. This will generate all the icon exports and IconNames (Uppercase and with _) from inside the raw directory
7. The icon is automatically added to the storybook and ready to use
8. Done!