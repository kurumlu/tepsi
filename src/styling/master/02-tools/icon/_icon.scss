@mixin icon($position: before, $icon: false, $styles: true){
  @if $position == both {
    $position: 'before, &::after';
  }
  // Either a :before or :after pseudo-element, or both, defaulting to :before
  &::#{$position} {
    @if $icon {
      // A particular icon has been specified
      content: "#{map-get($t1-icons__map, $icon)}";
    }
    @if $styles {
      font-family: "tepsi-fontello", sans-serif;
      font-style: normal;
      font-weight: normal;
      speak: none;

      display: inline-block;
      text-decoration: inherit;
      width: 1em;
      text-align: center;

      /* For safety - reset parent styles, that can break glyph codes*/
      font-variant: normal;
      text-transform: none;

      /* fix buttons height, for twitter bootstrap */
      line-height: 1em;

      /* Font smoothing. That was taken from TWBS */
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
    }
    // Include any extra rules supplied for the pseudo-element
    @content;
  }
}

@mixin icon-color($foreground) {
  .sd-a-icon__svg {
    fill: $foreground;
  }
}