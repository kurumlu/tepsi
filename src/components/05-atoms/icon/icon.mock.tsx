import { IconProps, IconName, IconColor, IconSize } from './Icon';

export const mockIconProps: IconProps = {
  name: IconName.MENU,
  className: 'sd-a-icon-test',
  color: IconColor.POSITIVE,
  size: IconSize.MAJOR,
};
