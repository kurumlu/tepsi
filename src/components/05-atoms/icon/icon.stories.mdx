import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { iconShowCase } from '../../../storybook_addons/storyWrapperDecorator';
import { Icon, IconName, IconColor, IconSize, IconAnimation } from './Icon';
import { Text, TextSize, TextColor } from '../text/Text';


<Meta
  title="Design system/Patterns/Atoms/Icon"
  component={Icon}
  decorators={[
    (Story) => (
        <Story />
    ),
  ]}
/>

# Icon

<Canvas>
  <Story name="All">
    {Object.values(IconName).map((name, i) => (
      <div style={{ width: 300 }}>
          <Icon title={`IconName.${Object.keys(IconName)[i]}`} name={name} key={i} />
          <Text size={TextSize.SMALL} color={TextColor.SUBTLE}>{Object.keys(IconName)[i]}</Text>
      </div>
    ))}
  </Story>
</Canvas>

---

## Available props

<ArgsTable for={Icon} />

---

### Name

Use the `name` prop to choose an icon. Uses a component from `IconName`.

```javascript
<Icon name={...} />
```

### Color

Use the `color` prop to change the color of the icon. Uses the `IconColor` enum.

```javascript
<Icon color={...} />
```

<Canvas>
  <Story name="Color">
    <Icon name={IconName.STORE} color={IconColor.PRIMARY} />
    <Icon name={IconName.STORE} color={IconColor.SECONDARY} />
    <Icon name={IconName.STORE} color={IconColor.TERTIARY} />
    <Icon name={IconName.STORE} color={IconColor.ACTION} />
    <Icon name={IconName.STORE} color={IconColor.ACTION_NAV} />
    <Icon name={IconName.STORE} color={IconColor.NEGATIVE} />
    <Icon name={IconName.STORE} color={IconColor.POSITIVE} />
    <Icon name={IconName.STORE} color={IconColor.SUBTLE} />
    <Icon name={IconName.STORE} color={IconColor.NOTICE} />
  </Story>
</Canvas>


### Size

Use the `size` prop to change the size of the icon. Uses the `IconSize` enum.

```javascript
<Icon size={...} />
```

<Canvas>
  <Story name="Size">
    <Icon name={IconName.GIFT} size={IconSize.EXTRA_MINOR} />
    <Icon name={IconName.GIFT} size={IconSize.MINOR} />
    <Icon name={IconName.GIFT} size={IconSize.MEDIUM} />
    <Icon name={IconName.GIFT} size={IconSize.MAJOR} />
    <Icon name={IconName.GIFT} size={IconSize.EXTRA_MAJOR} />
  </Story>
</Canvas>


### Animation

Use the `animation` prop to set the special type of the animation for icon. Uses the `IconAnimation` enum.

```javascript
<Icon animation={...} />
```

<Canvas>
  <Story name="Animation">
    <Icon name={IconName.LOCATION} animation={IconAnimation.BOP} />
    <Icon name={IconName.LOCATION} animation={IconAnimation.HEART_PULSE} />
    <Icon name={IconName.LOCATION} animation={IconAnimation.SPIN} />
  </Story>
</Canvas>


### Other

Other props that can be added:
- `id`
- `className`
- `dynamicContent`
- `ariaLabel`
- `ariaHidden`
- `inline`
