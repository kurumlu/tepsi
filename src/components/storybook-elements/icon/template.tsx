import * as React from 'react';
import { Icon, IconName } from '../../05-atoms/icon/Icon';

export const Icons = (
  <>
    <h5>Icons</h5>
    <div className="storybook-style-icons-wrapper">
      <b className="storybook-style-icons-wrapper__title">Social media:</b>
      <ul className="storybook-style-icons">
        <li className="storybook-style-icons__item">
          <Icon name={IconName.LINKEDIN_BOX} />
          <div className="storybook-style-icons__text">linkedin-box</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.FACEBOOK_BOX} />
          <div className="storybook-style-icons__text">facebook-box</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.GOOGLE_PLUS_BOX} />
          <div className="storybook-style-icons__text">google-plus-box</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.INSTAGRAM} />
          <div className="storybook-style-icons__text">instagram</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.TWITTER_BOX} />
          <div className="storybook-style-icons__text">twitter-box</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.PINTEREST_BOX} />
          <div className="storybook-style-icons__text">pinterest-box</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.AMAZON} />
          <div className="storybook-style-icons__text">amazon</div>
        </li>
      </ul>
    </div>
    <div className="storybook-style-icons-wrapper">
      <b className="storybook-style-icons-wrapper__title">Customer/ account:</b>
      <ul className="storybook-style-icons">
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MESSAGE} />
          <div className="storybook-style-icons__text">message</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.ACCOUNT_MULTIPLE} />
          <div className="storybook-style-icons__text">account-multiple</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.PHONE} />
          <div className="storybook-style-icons__text">phone</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.INFORMATION_OUTLINE} />
          <div className="storybook-style-icons__text">information-outline</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.EMAIL} />
          <div className="storybook-style-icons__text">email</div>
        </li>
      </ul>
    </div>
    <div className="storybook-style-icons-wrapper">
      <b className="storybook-style-icons-wrapper__title">
        Delivery/ checkout:
      </b>
      <ul className="storybook-style-icons">
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CALENDAR_TEXT_OUTLINE} />
          <div className="storybook-style-icons__text">
            calendar-text-outline
          </div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.TRUCK} />
          <div className="storybook-style-icons__text">truck</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.TRUCK_REVERSE} />
          <div className="storybook-style-icons__text">truck reverse</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CART} />
          <div className="storybook-style-icons__text">cart</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.HEART} />
          <div className="storybook-style-icons__text">heart</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.HEART_EMPTY} />
          <div className="storybook-style-icons__text">heart empty</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.STORE} />
          <div className="storybook-style-icons__text">store</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.STAR} />
          <div className="storybook-style-icons__text">star</div>
        </li>
      </ul>
    </div>
    <div className="storybook-style-icons-wrapper">
      <b className="storybook-style-icons-wrapper__title">
        Delivery/ checkout:
      </b>
      <ul className="storybook-style-icons">
        <li className="storybook-style-icons__item">
          <Icon name={IconName.ARROW_LEFT} />
          <div className="storybook-style-icons__text">arrow-left</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.ARROW_RIGHT} />
          <div className="storybook-style-icons__text">arrow-right</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.ARROW_TOP_LEFT} />
          <div className="storybook-style-icons__text">arrow-top-left</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.ARROW_TOP_RIGHT} />
          <div className="storybook-style-icons__text">arrow-top-right</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.REDO_SOLID} />
          <div className="storybook-style-icons__text">redo-solid</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CHEVRON_LEFT} />
          <div className="storybook-style-icons__text">chevron-left</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CHEVRON_RIGHT} />
          <div className="storybook-style-icons__text">chevron-right</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CHEVRON_UP} />
          <div className="storybook-style-icons__text">chevron-up</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CHEVRON_DOWN} />
          <div className="storybook-style-icons__text">chevron-down</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.GIFT_SOLID} />
          <div className="storybook-style-icons__text">gift-solid</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.SUBDIRECTORY_ARROW_RIGHT} />
          <div className="storybook-style-icons__text">
            subdirectory-arrow-right
          </div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.LEFT} />
          <div className="storybook-style-icons__text">left</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.RIGHT} />
          <div className="storybook-style-icons__text">right</div>
        </li>
      </ul>
    </div>
    <div className="storybook-style-icons-wrapper">
      <b className="storybook-style-icons-wrapper__title">General:</b>
      <ul className="storybook-style-icons">
        <li className="storybook-style-icons__item">
          <Icon name={IconName.BOOK} />
          <div className="storybook-style-icons__text">book</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MENU} />
          <div className="storybook-style-icons__text">menu</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MAGNIFY_PLUS_OUTLINE} />
          <div className="storybook-style-icons__text">
            magnify-plus-outline
          </div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MAGNIFY_MINUS_OUTLINE} />
          <div className="storybook-style-icons__text">
            magnify-minus-outline
          </div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.VIEW_LIST} />
          <div className="storybook-style-icons__text">view-list</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.GLOBE} />
          <div className="storybook-style-icons__text">globe</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CHECK} />
          <div className="storybook-style-icons__text">check</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.PLAY_CIRCLE} />
          <div className="storybook-style-icons__text">play-circle</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.PLAY_CIRCLED} />
          <div className="storybook-style-icons__text">play-circled2</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.PENCIL} />
          <div className="storybook-style-icons__text">pencil</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.FILTER} />
          <div className="storybook-style-icons__text">filter</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.EYE_OFF} />
          <div className="storybook-style-icons__text">eye-off</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.EYE_OUTLINE} />
          <div className="storybook-style-icons__text">eye-outline</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.EXCLAMATION_CIRCLE_SOLID} />
          <div className="storybook-style-icons__text">
            exclamation-circle-solid
          </div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.EXCLAMATION_TRIANGLE_SOLID} />
          <div className="storybook-style-icons__text">
            exclamation-triangle-solid
          </div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.USER_SOLID} />
          <div className="storybook-style-icons__text">user-solid</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.USER_CHECK_SOLID} />
          <div className="storybook-style-icons__text">user-check-solid</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CLOSE} />
          <div className="storybook-style-icons__text">close</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.PLUS} />
          <div className="storybook-style-icons__text">plus</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MINUS} />
          <div className="storybook-style-icons__text">minus</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.BELL_SOLID} />
          <div className="storybook-style-icons__text">bell</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.HOME} />
          <div className="storybook-style-icons__text">home</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.COMMERCIAL_BUILDING} />
          <div className="storybook-style-icons__text">commercial-building</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CIRCLE} />
          <div className="storybook-style-icons__text">circle</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.COOKIE_BITE_SOLID} />
          <div className="storybook-style-icons__text">cookie</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.T_SHIRT_SOLID} />
          <div className="storybook-style-icons__text">t-shirt</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.FIT_FINDER} />
          <div className="storybook-style-icons__text">fit-finder</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.TRASH} />
          <div className="storybook-style-icons__text">trash</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CREDIT_CARD} />
          <div className="storybook-style-icons__text">credit card</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.CREDIT_CARD_CVC} />
          <div className="storybook-style-icons__text">credit card cvc</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.DOWNLOAD} />
          <div className="storybook-style-icons__text">download</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.LIGHT_BULB} />
          <div className="storybook-style-icons__text">light bulb</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.LEAF} />
          <div className="storybook-style-icons__text">leaf</div>
        </li>
      </ul>
    </div>
    <div className="storybook-style-icons-wrapper">
      <b className="storybook-style-icons-wrapper__title">Location:</b>
      <ul className="storybook-style-icons">
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MAP_MARKER} />
          <div className="storybook-style-icons__text">map-marker</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.LOCATION} />
          <div className="storybook-style-icons__text">location</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MAGNIFY} />
          <div className="storybook-style-icons__text">magnify</div>
        </li>
      </ul>
    </div>
    <div className="storybook-style-icons-wrapper">
      <b className="storybook-style-icons-wrapper__title">Gender:</b>
      <ul className="storybook-style-icons">
        <li className="storybook-style-icons__item">
          <Icon name={IconName.FEMALE} />
          <div className="storybook-style-icons__text">female</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.MALE} />
          <div className="storybook-style-icons__text">male</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.UNISEX} />
          <div className="storybook-style-icons__text">unisex</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.GIRL} />
          <div className="storybook-style-icons__text">gender-girl</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.BOY} />
          <div className="storybook-style-icons__text">gender-boy</div>
        </li>
        <li className="storybook-style-icons__item">
          <Icon name={IconName.JUNIOR} />
          <div className="storybook-style-icons__text">junior</div>
        </li>
      </ul>
    </div>
  </>
);
