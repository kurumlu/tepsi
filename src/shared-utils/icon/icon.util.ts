import { IconName } from '../../components/05-atoms/icon/Icon';

export const getIconCode = (id: number | string): IconName | null => {
  switch (String(id)) {
    case 'dms':
    case '2669':
      return IconName.FEMALE;
    case 'girl':
    case '2617':
      return IconName.GIRL;
    case 'hrn':
    case '2668':
      return IconName.MALE;
    case 'boy':
    case '2616':
      return IconName.BOY;
    case 'uni':
    case '2670':
      return IconName.UNISEX;
    case 'uni_jr':
    case '2618':
      return IconName.JUNIOR;
    default:
      return null;
  }
};
