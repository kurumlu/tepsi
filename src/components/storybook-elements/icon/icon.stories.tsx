import { storiesOf } from '@storybook/react';
import { Icons } from './template';
import { withPageContainerAndPadding } from '../../../storybook_addons/storyWrapperDecorator';

storiesOf('Design system/Style/Icons', module)
  .addDecorator(withPageContainerAndPadding)
  .add('all', () => Icons);
