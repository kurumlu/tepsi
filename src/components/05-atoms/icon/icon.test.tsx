import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Icon, IconProps, IconName } from './Icon';
import { mockIconProps } from './icon.mock';

describe('<Icon />', () => {
  it('renders with default props', () => {
    const component: ShallowWrapper<IconProps> = shallow(
      <Icon name={IconName.HEART} />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders with props', () => {
    const component: ShallowWrapper<IconProps> = shallow(
      <Icon {...mockIconProps} />
    );

    expect(component).toMatchSnapshot();
  });
});
