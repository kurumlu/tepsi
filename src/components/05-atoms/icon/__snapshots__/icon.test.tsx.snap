// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders with default props 1`] = `
<span
  className="sd-a-icon sd-a-icon--heart sd-a-icon--medium"
  role="img"
>
  <svg
    aria-hidden="true"
    className="sd-a-icon__svg"
    focusable="false"
  />
</span>
`;

exports[`<Icon /> renders with props 1`] = `
<span
  className="sd-a-icon sd-a-icon--menu sd-a-icon--positive sd-a-icon--major sd-a-icon-test"
  role="img"
>
  <svg
    aria-hidden="true"
    className="sd-a-icon__svg"
    focusable="false"
  />
</span>
`;
